integrations/access/opsgenie/config.go

```diff
@@ -38,6 +39,8 @@ type Config struct {
 	AccessTokenProvider auth.AccessTokenProvider
 	// StatusSink defines a destination for PluginStatus
 	StatusSink common.StatusSink
+	// Ctx is the context used for the lifetime of the opsgenie plugin.
+	Ctx context.Context
```

Comment:
It's pretty weird to store a `context.Context` on a struct like this - and I'm not sure this comment actually describes what it does.